contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -37,6 +118,81 @@ contract TeleporterRegistry is WarpProtocolRegistry {
         view
         returns (ITeleporterMessenger)
     {
-        return ITeleporterMessenger(_getAddressFromVersion(_latestVersion));
+        return ITeleporterMessenger(getAddressFromVersion(latestVersion));
+    }
+
+    /**
+     * @dev Gets the {ITeleporterMessenger} contract of the given `version`.
+     * Requirements:
+     *
+     * - `version` must be a valid version, i.e. greater than 0 and not greater than the latest version.
+     */
+    function getTeleporterFromVersion(
+        uint256 version
+    ) external view returns (ITeleporterMessenger) {
+        return ITeleporterMessenger(getAddressFromVersion(version));
+    }
+
+    /**
+     * @dev Gets the address of a protocol version.
+     * Requirements:
+     *
+     * - `version` must be a valid version, i.e. greater than 0 and not greater than the latest version.
+     */
+    function getAddressFromVersion(
+        uint256 version
+    ) public view returns (address) {
+        require(version != 0, "TeleporterRegistry: zero version");
+        require(
+            version <= latestVersion,
+            "TeleporterRegistry: invalid version"
+        );
+        return _versionToAddress[version];
+    }
+
+    /**
+     * @dev Gets the version of the given `protocolAddress`.
+     * If `protocolAddress` is not a registered protocol address, returns 0, which is an invalid version.
```

Comment:
In `getAddressFromVersion`, we revert if an invalid version is provided, but here we return 0 if an invalid address is provided. Do we want to revert in this case as well?

Reply:
hmm ye I thought about this too, but `getAddressFromVersion` is used to check for `onlyAllowedTeleporter` in `TeleporterUpgradeable` as well. If we add an extra revert here that'd still be okay for the modifier, but it'd revert with a different error message. No strong pref here, thoughts?

Added a check that the parameter `protocolAddress` is not zero address.

Reply:
My preference would be to have the same pattern of behavior for both methods in the event they are given either a version or address that isn't set. No strong preference between them both reverting or both returning the default value (`0` or `address(0)`) in those cases though.

Reply:
checked it so that both functions check for registered inputs, and reverts if the input given is not registered.